src/app/onboarding/review/page.tsx

```diff
@@ -0,0 +1,227 @@
+'use client';
+
+import { useState } from 'react';
+import Link from 'next/link';
+import { useRouter } from 'next/navigation';
+import { BigBlueAsyncButton, BigLinkButton } from '@/components/Buttons';
+import Icon from '@/components/Icon';
+import COLORS from '@/styles/colors';
+import { Card, Flex } from '@/styles/containers';
+import { H1Centered, H2, H4, P } from '@/styles/text';
+import { formatTruthy, getUTCDate, parseDate } from '@/utils/helpers';
+import { useGuardedOnboarding, useOnboardingNavigation } from '@/utils/hooks';
+import * as Styles from '../styles';
+
+function EditButton({ href }: { href: string }) {
+  return (
+    <Link href={href}>
+      <Flex $gap="5px">
+        <Styles.EditText>Edit</Styles.EditText>
+        <Icon type="edit" />
+      </Flex>
+    </Link>
+  );
+}
+
+export default function Page() {
```

Comment:
i went through the whole onboarding flow without being logged in, and it only displayed an error on this page, and it was just for a second before successfully moving into the final screen that directs the user back to home.

some things here:
1. should we stop them earlier if they are not logged in?
2. this error should prevent the final success screen from being shown altogether, right?

Reply:
you're so right
ill make it so it waits for flush data first